### ISSUE TITLE
Sumac Sandbox & Release Preparation

### ISSUE THREAD

**Issue by DawoudSheraz** [overhangio/tutor#1121]
This issue is for tracking the items to prepare https://github.com/overhangio/openedx-release-demo for Sumac cutoff.

Sumac cutoff is expected to take place on **October 23, 2024**. Once the branches are out on upstream, the community will be looking forward to sandbox provided by Edly. tutor and its plugins will require `sumac` branches with appropriate changes for the sandbox to complete. The necessary information is already present on https://edlyio.atlassian.net/wiki/spaces/openedx/pages/3013148726/Building+Open+edX+major+releases.  **A note: once the branches are created, any changes to nightly would need to be manually added to the sumac branch. Therefore, a rebase on nightly whenever nightly changes would be required to ensure the sumac branch is always up-to-date**

This issue is meant for tracking the statuses of various repositories needed for sandbox. It will act as a todo list to ensure that tutor and its plugins have sumac branches created from nightly before the cutoff date (even though there will be potential failures on CI against it) and have a PR against master. 
For sanity checks, make sure to run image build and init task on local `tutor <local/dev> do init --limit=pluginnname`. 

**The sandbox script expects the branches to be available on the main repository, not the fork. If you are assigned a repository but you cannot push to it directly, please get in touch with @DawoudSheraz or the maintainer of the repository with patch file so that they can create the PR in the appropriate repository with your provided changes** 

#### Available on launch
- [x] Tutor sumac PR (https://github.com/overhangio/tutor/pull/1139 by @DawoudSheraz )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-mfe sumac PR (https://github.com/overhangio/tutor-mfe/pull/227 by @hinakhadim )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-indigo sumac PR (https://github.com/overhangio/tutor-indigo/pull/101 by @hinakhadim )
  - [x] Is nightly local image build working?
  - [x] Is init working?

#### Available within a week
- [x] tutor-discovery sumac PR (https://github.com/overhangio/tutor-discovery/pull/87 by @Faraz32123 )
  - [x] Is nightly local image build working? 
  - [x] Is init working?
- [x] tutor-ecommerce sumac PR (https://github.com/overhangio/tutor-ecommerce/pull/84 by @Faraz32123 )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-credentials sumac PR (https://github.com/overhangio/tutor-credentials/pull/49 by @Faraz32123 )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-forums sumac PR (https://github.com/overhangio/tutor-forum/pull/49 by @ghassanmas )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-notes sumac PR (https://github.com/overhangio/tutor-notes/pull/41 by @jfavellar90 )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-xqueue sumac PR (https://github.com/overhangio/tutor-xqueue/pull/34 by @jfavellar90 )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-jupyter sumac PR (https://github.com/overhangio/tutor-jupyter/pull/15 by @Abdul-Muqadim-Arbisoft )
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-minio sumac PR (https://github.com/overhangio/tutor-minio/pull/51 by @Faraz32123  )
  - [x] Is init working?
- [x] tutor-webui sumac PR (https://github.com/overhangio/tutor-webui/pull/19 by @Abdul-Muqadim-Arbisoft & @hinakhadim)
  - [x] Is nightly local image build working?
  - [x] Is init working?
- [x] tutor-android sumac PR (https://github.com/overhangio/tutor-android/pull/28 by @muhammadali286 & @Abdul-Muqadim-Arbisoft)
   - [x] Is nightly local image build working?
   - [x] Is init working?
- [x] tutor-contrib-codejail PR (https://github.com/eduNEXT/tutor-contrib-codejail/pull/59 by @MoisesGSalas)
   - [x] Is nightly local image build working?
   - [x] Is init working?
- [x] tutor-contrib-aspects v1.2.0

#### Not needed for sandbox
- [x] tutor-cairn sumac PR (https://github.com/overhangio/tutor-cairn/pull/47/files by @Danyal-Faheem  )
  - [x] Is nightly local image build working?
  - [x] Is init working?

### Sandbox Repo Todo List
- [x] sumac preparation PR (https://github.com/overhangio/openedx-release-demo/pull/61 by @DawoudSheraz )
- [x] Enable forum https://github.com/overhangio/openedx-release-demo/pull/62
- [x] Enable some of the remaining plugins https://github.com/overhangio/openedx-release-demo/pull/63 
- [x] Enable the last set of plugins https://github.com/overhangio/openedx-release-demo/pull/64 

### Context
Every six months, Tutor maintainers sync up with the Build/Test/Release working group to create the next version, both of Open edX and Tutor.

In the scenario below, we are upgrading Open edX from fictional “Delta” to “Epsilon”. Tutor will upgrade from v4 to v5  (“d”=4, “e”=5).

(these instructions are pulled and adapted from [this old discussion topic](https://discuss.overhang.io/t/how-to-upgrade-a-tutor-plugin/1488))

#### Create release branches
In Tutor core and all plugins, we must create “epsilon” branches off of the nightly branches. For each repo, this looks like the following:

```
git checkout nightly

git pull

git checkout -b epsilon

Push the created “epsilon” branch to the upstream repo: git push
```

#### Upgrade Tutor core and plugins
- Modify __about__.py:
  - bump the version number to 5.0.0.
- in Tutor core, set the __version_suffix__ to an empty string.
- In plugins’ setup.py, bump the version of the “tutor” package that the plugin depends on: 
```
install_requires=["tutor>=5.0.0,<6.0.0"],
extras_require={"dev": ["tutor[dev]>=5.0.0,<6.0.0"]},
```
- Create a changelog entry with make changelog-entry: "💥[Feature] Upgrade to Epsilon. (by @YOURGITHUBUSERNAME)"
- Replace all instances of “[dD]elta” that make sense (i.e: not in the CHANGELOG.md!)
- Go through the Dockerfile templates and manually upgrade all the 3rd-party requirements that you can find (dockerize, ipdb, etc.). Sometimes it’s not desirable to upgrade some pieces of software: for instance node and python are expected to run a specific version. Those required version numbers are specified in the edx/configuration 5 repo. (look for SERVICENAME_VERSION variables)
- Make sure that the plugin images (if any) are built correctly: tutor images build all.
- Make sure that the plugin can be correctly initialized: tutor local do init --limit=pluginnname.
- If you need to make some changes to the docker-compose*.yml files and patches, make sure to backport these changes to the k8s-* patches, for compatibility with Kubernetes.
- Make sure that the plugins work correctly by doing some basic usage testing. Pay close attention to any log (warning or error) that come out of the plugin containers as well as the lms container.
- git commit -a -m v5.0.0
- Push the created “epsilon” branch to the upstream repo: git push

The release branches should be updated regularly to take into account the latest changes from the nightly branch. During the release process, it is frequent that changes are pushed to the master and nightly branches, and the latest “espilon” branch must have those changes as well. Be prepared to push --force the “espsilon” branches frequently.

:warning: Plugin tests will fail in the release branches, because the plugins can’t find tutor>=5.0.0 on pypi. To resolve that, we should update the test scripts to install tutor from source (pip install -e --config-settings editable_mode=compat https://github.com/overhangio/tutor@epsilon#egg=tutor). But it’s inconvenient to do that for each and every plugin. So maybe we should consider migrating to a centralized “tutor-test-plugin” GitHub action. This action would be versioned according to the Tutor major version (@5).

**Comment by DawoudSheraz** [comment]
@jfavellar90 Hi, letting you know since you are the maintainer for notes and xqueue.

@ghassanmas Hi, letting you know since you are the maintainer for forum.

@MoisesGSalas Hi. I am not sure if you are the contact person for codejail but since you did it last time for redwood, I have assigned the sumac to you. Let me know if this is not the case. 

Thank you all. Let me know if you have any questions

**Comment by MoisesGSalas** [comment]
Hi @DawoudSheraz, that's right, I will take care of the codejail plugin.

**Comment by DawoudSheraz** [comment]
> Hi @DawoudSheraz, that's right, I will take care of the codejail plugin.

Awesome, thank you.

**Comment by jfavellar90** [comment]
@DawoudSheraz I'm taking care of notes and xqueue :)

**Comment by DawoudSheraz** [comment]
> @DawoudSheraz I'm taking care of notes and xqueue :)

Thank you.

**Comment by angonz** [comment]
Hi!
There are a number of tests for Aspects. Shall we include the [aspects plugin](https://github.com/openedx/tutor-contrib-aspects) in the list? Who is in charge?

**Comment by DawoudSheraz** [comment]
> Hi! There are a number of tests for Aspects. Shall we include the [aspects plugin](https://github.com/openedx/tutor-contrib-aspects) in the list? Who is in charge?

Hi, I am in touch with BTR to get a confirmation on this. Thanks

**Comment by angonz** [comment]
Hi @bmtcril! Can you please check if the aspects plugin is ready for the summac sandbox?

**Comment by bmtcril** [comment]
@angonz @DawoudSheraz we're working on the last couple of Aspects bug fixes, but expect to have a release ready in the next day or so.

**Comment by hinakhadim** [comment]
[Tutor-mfe](https://github.com/overhangio/tutor-mfe/pull/227) and [tutor-indigo](https://github.com/overhangio/tutor-indigo/pull/101) Sumac PRs are ready. I've tested them on local and sandboxdev using [tutor Sumac PR](https://github.com/overhangio/tutor/pull/1139) (with nightly)

**Comment by Faraz32123** [comment]
tutor-discovery sumac [PR](https://github.com/overhangio/tutor-discovery/pull/87)
- [x] Is nightly local image build working?
- [x]  Is init working?

 tutor-ecommerce sumac [PR](https://github.com/overhangio/tutor-ecommerce/pull/84)
- [x] Is nightly local image build working?
- [x]  Is init working?

 tutor-credentials sumac [PR](https://github.com/overhangio/tutor-credentials/pull/49)
- [x] Is nightly local image build working?
- [x]  Is init working?

**Comment by crathbun428** [comment]
@angonz @DawoudSheraz - Here is configuration information for Libraries for the sandbox: [](https://github.com/orgs/openedx/projects/66/views/1?filterQuery=config&pane=issue&itemId=81265913&issue=openedx%7Cfrontend-app-authoring%7C1334). Thank you!

**Comment by crathbun428** [comment]
@angonz @DawoudSheraz - Linking the configuration information for Libraries for the testing sandbox, [here]:([url](https://github.com/orgs/openedx/projects/66/views/1?filterQuery=config&pane=issue&itemId=81265913&issue=openedx%7Cfrontend-app-authoring%7C1334))

Thank you!

**Comment by jfavellar90** [comment]
tutor-notes sumac [PR](https://github.com/overhangio/tutor-notes/pull/41) is ready

- [ ]  Is nightly local image build working?
- [ ]  Is init working?

tutor-xqueue sumac [PR](https://github.com/overhangio/tutor-xqueue/pull/34) is ready

- [ ]  Is nightly local image build working?
- [ ]  Is init working?

**Comment by DawoudSheraz** [comment]
> @angonz @DawoudSheraz Aspects v1.2.0 is on PyPI and ready for testing, thanks for your patience!

Cool. I will update sandbox script to point to v1.2.0 tag for contrib-aspects. Thanks

**Comment by DawoudSheraz** [comment]
> @angonz @DawoudSheraz - Linking the configuration information for Libraries for the testing sandbox:[here](https://github.com/orgs/openedx/projects/66/views/1?filterQuery=config&pane=issue&itemId=81265913&issue=openedx%7Cfrontend-app-authoring%7C1334)
> 
> Thank you!

Hi. meilisearch related PRs are not merged yet, they are under review. I have not looked at the entire linked issue but it seems we need meilisearch before enabling the flags. @kdmccormick Is this correct?

**Comment by ghassanmas** [comment]
Tutor forum PRs are ready I have tested (v1) with discussion MFE. 
- tutor-forum https://github.com/overhangio/tutor-forum/pull/49 
- Demo PR https://github.com/overhangio/openedx-release-demo/pull/62 

Note: The discussion MFE and forum cs_comments_service would crash when responding to a thread, if you are disk is almost full, this because Elasticsearch blocks writing in such case.

**Comment by MoisesGSalas** [comment]
Tutor contrib codejail: https://github.com/eduNEXT/tutor-contrib-codejail/pull/59, still wip but is working.

- [x] Is nightly local image build working?
- [x] Is init working?

**Comment by DawoudSheraz** [comment]
> Tutor forum PRs are ready I have tested (v1) with discussion MFE.
> 
> * tutor-forum [feat: upgrade to sumac tutor-forum#49](https://github.com/overhangio/tutor-forum/pull/49)
> * Demo PR [feat: add forum versoin1 openedx-release-demo#62](https://github.com/overhangio/openedx-release-demo/pull/62)
> 
> Note: The discussion MFE and forum cs_comments_service would crash when responding to a thread; if your disk is almost full, this because Elasticsearch blocks writing in such case.

Thanks

**Comment by Faraz32123** [comment]
tutor-minio sumac [PR](https://github.com/overhangio/tutor-minio/pull/51) is ready.
- [x] Is init working?

**Comment by Danyal-Faheem** [comment]
tutor-cairn sumac [PR](https://github.com/overhangio/tutor-cairn/pull/47/files) is ready.

- [x] Is nightly local image build working?
- [x] Is init working?

**Comment by angonz** [comment]
Hi team! Can we confirm that the basic sandbox is running, so we can start with the tests?

**Comment by Abdul-Muqadim-Arbisoft** [comment]
tutor-jupyter sumac [PR](https://github.com/overhangio/tutor-jupyter/pull/15) is ready.

- [x]  Is nightly local image build working?
- [x]  Is init working?

tutor-webui sumac [PR](https://github.com/overhangio/tutor-webui/pull/19) is ready.

- [x]  Is nightly local image build working?
- [x]  Is init working?

**Comment by angonz** [comment]
Hi @DawoudSheraz,
I got some complaints that testers don't receive emails. There is a [reported issue](https://openedx.slack.com/archives/C049JQZFR5E/p1716298199246939) with some email providers, probably due to AWS not allowing port 25 out of the server.
Fayyaz Ahmed reported that emails from arbisoft.com don't work either. CC @mariajgrimaldi who also worked on that. Can we review this? Which SMTP server is using the sandbox?

**Comment by DawoudSheraz** [comment]
> Hi @DawoudSheraz, I got some complaints that testers don't receive emails. There is a [reported issue](https://openedx.slack.com/archives/C049JQZFR5E/p1716298199246939) with some email providers, probably due to AWS not allowing port 25 out of the server. Fayyaz Ahmed reported that emails from arbisoft.com don't work either. CC @mariajgrimaldi who also worked on that. Can we review this? Which SMTP server is using the sandbox?

Hello. This happened with redwood as well. Port 25 is blocked on the instance, this is by default by AWS. We did not update any spam-related records because the sandbox is public and anyone can send emails from that (by sign up, activation, etc.). It risks impacting the domain authority. 
In Redwood testing, it was decided to update testing instructions to highlight using self hosted email accounts for getting emails https://github.com/openedx/wg-build-test-release/issues/350#issuecomment-2133772454. 

@regisb Please add any context that I might have missed.

**Comment by mariajgrimaldi** [comment]
@angonz: As mentioned in the thread, emails won't work for email providers like Google, Yahoo, etc. However, I tested with proton mail, which worked for me for the Redwood testing, and it didn't work either.

**Comment by Abdul-Muqadim-Arbisoft** [comment]
tutor-android sumac [PR](https://github.com/overhangio/tutor-android/pull/28) is ready.

- [x]  Is nightly local image build working?
- [x]  Is init working?

**Comment by DawoudSheraz** [comment]
> @angonz @DawoudSheraz: As mentioned in the thread, emails won't work for email providers like Google, Yahoo, etc. I'll try with the protonmail account that I used during the Redwood release; if it doesn't work, I'll use a colleague with a self-hosted email account to test.

@mariajgrimaldi Hi, can you verify this again? It should be set now.

**Comment by mariajgrimaldi** [comment]
@DawoudSheraz it works with my protomain account! Thank you. @angonz: I think we should inform the tester what provider might work, like protonmail.

**Comment by DawoudSheraz** [comment]
> The new Libraries feature is not showing up in the sandbox - these are the configuration requirements, please let me know if I can help. [openedx/frontend-app-authoring#1334](https://github.com/openedx/frontend-app-authoring/issues/1334)

meilisearch PRs are not part of sumac branches yet.

**Comment by DawoudSheraz** [comment]
Some updates on sandbox:
- Meilisearch PRs have been merged and deployed as part of today's sandbox re-run. Libraries v2 interface is now accessible on Studio. cc: @jmakowski1123 @crathbun428 
- tutor-forum has been disabled temporarily because of some merge conflicts issues of forum_v2 branch https://github.com/overhangio/openedx-release-demo/pull/65. Once those issues are resolved, tutor-forum will be re-enabled. cc: @angonz 


Let me know if there are any questions or concerns.

**Comment by DawoudSheraz** [comment]
v19 is now available for tutor core and plugins. Closing this issue. 
See ya for Teak.
